truss/templates/control/control/endpoints.py

```diff
@@ -25,12 +27,24 @@ def proxy(path):
         wait=wait_fixed(1),
     ):
         with attempt:
-            resp = requests.request(
-                method=request.method,
-                url=f"http://localhost:{inference_server_port}/v1/{path}",
-                data=request.get_data(),
-                cookies=request.cookies,
-            )
+            try:
+                resp = requests.request(
+                    method=request.method,
+                    url=f"http://localhost:{inference_server_port}/v1/{path}",
+                    data=request.get_data(),
+                    cookies=request.cookies,
+                )
+            except ConnectionError as exp:
+                # This check is a bit expensive so we don't do it before every request, we
+                # do it only if request fails with connection error. If the inference server
+                # process is running then we continue waiting for it to start (by retrying),
+                # otherwise we bail.
+                if not inference_server_process_controller.is_inference_server_running():
+                    return Response(
+                        'Inference server is not running, make sure model code doesn\'t have any errors',
```

Comment:
More a UX question, but possible for us to be more approachable with this message? Something like: "It appears your model has stopped running.  This often means it crashed and may need a fix to get it running again."

Reply:
Yes, that reads better, let me update, thx.